src/VisualStudio/Core/Def/Progression/GraphBuilder.cs

```diff
@@ -836,11 +837,11 @@ public Graph Graph
             }
         }
 
-        public IEnumerable<GraphNode> GetCreatedNodes(CancellationToken cancellationToken)
+        public ImmutableArray<GraphNode> GetCreatedNodes(CancellationToken cancellationToken)
```

Comment:
changed a bunch of code to be IMmutableArray to make it clear that lazy computation is not happening (especially scary with all the locks here).